### PR TITLE
Only pull Script Pod logs newer than the last set we got

### PR DESCRIPTION
**Pull request by flin-8**
[sc-71685]
# Background

We currently stream all logs from the Script Pod every time. There's a `sinceTime` filter we can use to only get the latest logs.

This PR adds `sinceTime` to the request we send. We also depend on the API Server's timestamp now instead of writing our own in the bootstrapper - in case the clocks are out of sync.

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/main/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: docker/kubernetes-tentacle/bootstrapRunner/bootstrapRunner.go [hunk 1]
```diff
@@ -88,7 +87,7 @@ func Write(stream string, text string, counter *SafeCounter) {
 	//https://go.dev/tour/concurrency/9
 	counter.Mutex.Lock()
 
-	fmt.Printf("%d|%s|%s|%s\n", counter.Value, time.Now().UTC().Format(time.RFC3339Nano), stream, text)
+	fmt.Printf("|%d|%s|%s\n", counter.Value, stream, text)
```
Comment (flin-8) [inline docker/kubernetes-tentacle/bootstrapRunner/bootstrapRunner.go:90]: Remove our own timestamp - rely on the API Server's timestamp instead (it gets added at the front with a space character

File: source/Octopus.Tentacle/Kubernetes/ScriptPodSinceTimeStore.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+namespace Octopus.Tentacle.Kubernetes
+{
+    public interface IScriptPodSinceTimeStore
+    {
+        DateTimeOffset? GetSinceTime(ScriptTicket scriptTicket);
+        void UpdateSinceTime(ScriptTicket scriptTicket, DateTimeOffset nextSinceTime);
+        void Delete(ScriptTicket scriptTicket);
+    }
+
+    public class ScriptPodSinceTimeStore : IScriptPodSinceTimeStore
```
Comment (flin-8) [inline source/Octopus.Tentacle/Kubernetes/ScriptPodSinceTimeStore.cs:14]: I decided to not put this on the tracked pods, this means the monitoring is not coupled to the log reading

File: docker/kubernetes-tentacle/bootstrapRunner/bootstrapRunner.go (hunk 1 above)
Comment (APErebus) [reply docker/kubernetes-tentacle/bootstrapRunner/bootstrapRunner.go:90]: is it in the correct format/resolution?

File: source/Octopus.Tentacle/Kubernetes/KubernetesClientExtensionMethods.cs [hunk 3]
```diff
@@ -4,21 +4,23 @@
...
+            //Include Server timestamps so we can use it to specify "sinceTime" on the next call
+            var url = $"api/v1/namespaces/{namespaceParameter}/pods/{podName}/log?container={container}&timestamps=true";
 
             if (sinceTime is not null)
             {
                 var sinceTimeStr = sinceTime.Value.ToString("O");
                 url += $"&sinceTime={Uri.EscapeDataString(sinceTimeStr)}";
             }
-
+
```
Comment (scme0) [inline source/Octopus.Tentacle/Kubernetes/KubernetesClientExtensionMethods.cs:23]: Nit: Remove whitespace please :)

File: docker/kubernetes-tentacle/bootstrapRunner/bootstrapRunner.go (hunk 1 above)
Comment (flin-8) [reply docker/kubernetes-tentacle/bootstrapRunner/bootstrapRunner.go:90]: Good question - looks like it - I have a temp build that prints "Parsing line: ..."

CleanShot 2024-04-10 at 17 22 44@2x

**Review by APErebus — APPROVED** [review]
Awesome stuff 👍

**Pull request merged**